### PR TITLE
add adafruit_feather_esp32s3_4mbflash_2mbpsram

### PR DESCRIPTION
**Pull request by dhalbert**
A mix of `adafruit_feather_esp32s3_tft` and `adafruit_feather_esp32s3_nopsram`, with name and USB PID changes.

### CODE REVIEW

File: ports/espressif/boards/adafruit_feather_esp32s3_4mbflash_2mbpsram/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// Micropython setup
+
+#define MICROPY_HW_BOARD_NAME       "Adafruit Feather ESP32S3 4MB Flash 2MB PSPRAM"
```
Comment (tannewt) [inline ports/espressif/boards/adafruit_feather_esp32s3_4mbflash_2mbpsram/mpconfigboard.h:29]:
```suggestion
#define MICROPY_HW_BOARD_NAME       "Adafruit Feather ESP32S3 4MB Flash 2MB PSRAM"
```

**Review by tannewt — CHANGES_REQUESTED** [review]
One typo. Good otherwise. Thanks!

File: ports/espressif/boards/adafruit_feather_esp32s3_4mbflash_2mbpsram/mpconfigboard.h (hunk 1 above)
Comment (dhalbert) [reply ports/espressif/boards/adafruit_feather_esp32s3_4mbflash_2mbpsram/mpconfigboard.h:29]: Thanks, will fix. I am getting one these tomorrow and will test it.

**Comment by dhalbert** [comment]
I received one of these boards yesterday and did an I2C sensor smoke test, which works.

**Comment by ladyada** [comment]
tested i2c, wifi all good!

**Review by ladyada — APPROVED** [review]

**Pull request merged**